### PR TITLE
fix for bug(issue  #2106)

### PR DESCRIPTION
**Pull request by shaipranesh2**
fixes issue #2106 , allows user to predict without labels.

### CODE REVIEW

File: deepchem/models/tests/test_graph_models.py [hunk 1]
```diff
@@ -353,3 +353,17 @@ def test_dtnn_regression_model():
   pred = model.predict(dataset)
   mean_rel_error = np.mean(np.abs(1 - pred / y))
   assert mean_rel_error < 0.1
+
+@pytest.mark.tensorflow
+def test_graph_predict():
+  model = dc.models.GraphConvModel(np.random.randint(1, 100), batch_size=100, mode='classification')    
+  mols = []
+  tasks, all_dataset, transformers = load_delaney(featurizer='GraphConv')
```
Comment (rbharath) [inline deepchem/models/tests/test_graph_models.py:361]: We should use a smaller synthetic dataset.

```
smiles = ["CCCCC", "CCCCCCCCC"]
y = [0, 1]
feat = dc.feat.ConvMolFeaturizer()
X = feat.featurize(smiles)
dataset = dc.data.NumpyDataset(X, y)
```

File: deepchem/models/tests/test_graph_models.py [hunk 2]
```diff
@@ -353,3 +353,17 @@ def test_dtnn_regression_model():
...
+  train, valid, test = all_dataset
+  molecules=train[:np.random.randint(2, 100)]
+  for X, y, w, id in train.itersamples():
+    mols.append(id)
+  feat = dc.feat.ConvMolFeaturizer()   
+  X = feat.featurize(mols)         
+  model.predict(dc.data.NumpyDataset(X))
```
Comment (rbharath) [inline deepchem/models/tests/test_graph_models.py:368]: We should have some assert statement

File: deepchem/models/graph_models.py [hunk 3]
```diff
@@ -1001,7 +1001,7 @@ def default_generator(self,
           batch_size=self.batch_size,
           deterministic=deterministic,
           pad_batches=pad_batches):
-        if y_b is not None and self.mode == 'classification':
+        if y_b is not None and self.mode == 'classification' and not(mode=='predict'):
```
Comment (rbharath) [inline deepchem/models/graph_models.py:1004]: Yapf is failing on this PR. To fix, install yapf
```
pip install yapf==0.32
```
Then run
```
yapf -i <filename>
```
on each modified file

**Review by rbharath — COMMENTED** [review]
Looking good but a few comments to address

File: deepchem/models/tests/test_graph_models.py [hunk 4]
```diff
@@ -353,3 +346,17 @@ def test_dtnn_regression_model():
   pred = model.predict(dataset)
   mean_rel_error = np.mean(np.abs(1 - pred / y))
   assert mean_rel_error < 0.1
+
+
+@pytest.mark.tensorflow
+def test_graph_predict():
+  model = dc.models.GraphConvModel(np.random.randint(1, 100),
```
Comment (rbharath) [inline deepchem/models/tests/test_graph_models.py:353]: Duplicated model definition

File: deepchem/models/tests/test_graph_models.py [hunk 5]
```diff
@@ -353,3 +346,17 @@ def test_dtnn_regression_model():
...
+                                   batch_size=100,
+                                   mode='classification')
+
+  model = dc.models.GraphConvModel(12, batch_size=50, mode='classification')
+  mols = ["CCCCC", "CCCCCCCCC"]
+  y = [0, 1]
+  feat = dc.feat.ConvMolFeaturizer()
+  X = feat.featurize(mols)
+  assert (model.predict(dc.data.NumpyDataset(X, y))).all() == True
```
Comment (rbharath) [inline deepchem/models/tests/test_graph_models.py:362]: Should this be
```
assert model.predict(dc.data.NumpyDataset(X))
```
If you want to check whether this works without tasks being labeled
```
assert model.predict(dc.data.NumpyDataset(X, y))
```
but that's a different check

**Review by rbharath — COMMENTED** [review]
A couple of minor feedback points.

**Comment by shaipranesh2** [comment]
@rbharath can you please review this PR

**Review by rbharath — APPROVED** [review]
Looks good I think. This is the first PR I've seen in a long time that has a green CI!

**Pull request merged**